### ISSUE TITLE
Provide conditional usage for `setupMiddlewareReporter`

### ISSUE THREAD

**Issue by drewlee** [ember-a11y/ember-a11y-testing#266]
When enabling ember-a11y-testing's [middleware reporter](https://github.com/ember-a11y/ember-a11y-testing#test-middleware), tests that normally fail from `a11yAudit` violations no longer do so. This technically makes sense, as the [default reporter is overridden](https://github.com/ember-a11y/ember-a11y-testing/blob/master/addon-test-support/setup-middleware-reporter.ts#L83) so that Errors are not thrown to collect the entirety of a11y violations. However, this requires host apps to add a bunch of setup code in `tests/test-helper.js` to interop between the default reporter and the middleware reporter. It's also not clear from the documentation on how to do this, or that this is expected behavior. For example:
```javascript
import { setupMiddlewareReporter } from 'ember-a11y-testing/test-support';

function someConditionToUseMiddlewareReporter() {
  // Some logic to determine which reporter to use
  // E.g., only use MR when running tests from CI
}

if (someConditionToUseMiddlewareReporter()) {
  setupMiddlewareReporter();
}
```

I think it's important to be able to interop between the two reporters, as it makes it difficult to otherwise debug test failures due to a lack of visible feedback. It's also a bit unintuitive that calling `setupMiddlewareReporter` basically hijacks the way ember-a11y-testing typically works. I propose that this addon provides a new low-level utility that allows host apps to run the middleware reporter conditionally. Then, the host app can simply provide a query parameter (or environmental variable?) to choose which reporter is used. For example:
```javascript
import { shouldUseMiddlewareReporter, setupMiddlewareReporter } from 'ember-a11y-testing/test-support';

if (shouldUseMiddlewareReporter()) {
  setupMiddlewareReporter();
}
```

If they don't require interop, then they can simply invoke `setupMiddlewareReporter` directly, as can be done today.

**Comment by drewlee** [comment]
I've also played around with just using an environmental variable and using a check as follows, although I'm not sure if this typical convention:
```javascript
// ember-config/environment.js
if (process.env.A11Y_MW_REPORTER) {
  ENV.A11Y_MW_REPORTER = true;
}

// tests/test-helper.js
import config from '../config/environment';
...

if (config.A11Y_MW_REPORTER) {
  setupMiddlewareReporter();
}
```
Not opposed to this approach, but would be helpful to document.

**Comment by scalvert** [comment]
This seems like a good addition. It won't require apps to manage this themselves by setting custom conditionals for tests.

We could actually support _both_ a qp _and_ an env var, but for both our conditionals. We could extend `enableA11yAudit` to be triggered from both, which would enable users to set these values either from CI (gh actions, travis.yml, etc) or via a qp.

I'd opt to standardize the naming a bit:

`enableA11yAudit` and/or `ENABLE_A11Y_AUDIT`
`enableA11yMiddlewareReporter` and/or `ENABLE_A11Y_MIDDLEWARE_REPORTER`

WDYT?

**Comment by drewlee** [comment]
Yeah, I like the idea of providing both options (query parameter and env variable) and agree with the naming standardization. The only tricky part would be figuring out how to pass off the environmental variables from the node process and make them accessible in the addon's browser code.

**Comment by drewlee** [comment]
Addressed with PR #267.
